### DISCUSSION TITLE
ControlNet support

### DISCUSSION THREAD

**Discussion by monstruosoft** [rupeshs/fastsdcpu#149] [General]
Much like LoRA, ControlNet is supported natively by the _diffusers_ library, so it should be possible to add ControlNet support to FastSD CPU.

![controlnet-1](https://github.com/rupeshs/fastsdcpu/assets/29698839/4d327462-8abf-4da1-8e06-6bf213e83e88)

**Comment by rupeshs** [comment]
Ipadapter is lightweight and better

**Comment by monstruosoft** [comment]
Gonna take a look into it.

**Comment by monstruosoft** [comment]
I'm looking at IP-Adapter; the adapter models are indeed smaller, but doesn't it also require loading the 2.5 GB _image_encoder_ model?.

**Reply by rupeshs** [reply]
yes to encode image, IP adpater uses a decoupled cross-attention mechanism so we need to get image features.
